soc/nordic/common/dmm.c

```diff
@@ -295,3 +295,10 @@ int dmm_init(void)
 
 	return 0;
 }
+
+/*
+ * dmm_init relies on kheap, initialized in PRE_KERNEL_1 at
+ * CONFIG_KERNEL_INIT_PRIORITY_OBJECTS. To enable early device use of
+ * dmm, we initialize immediately after kheap.
+ */
+SYS_INIT(dmm_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
use soc hook

Reply:
soc_early is too early, as dmm_init depends on kheap (PRE_KERNEL_1).

And soc_late is too late I believe, as device drivers initialized in PRE_KERNEL_1 use dmm.

I believe @anangl may be familiar with the details.

Also see commit message
https://github.com/zephyrproject-rtos/zephyr/commit/c84c2fc37dd0d76c5b6735ed72c25de073ea2993

Reply:
well you're going to have problems with this as sys_init is basically banned from boards and socs in zephyr

Reply:
Note that in this PR I am reducing reliance on SYS_INIT, so I feel a PR that moves us a step in the right direction doesn't deserve a -1.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/issues/75907 can ask @nashif for how to do this properly

Reply:
I recommend to move forward with Sebastian's solution. As dmm is part of low level work I will address it soon to establish compliant solution